demo/full/scripts/modules/bxf.js

```diff
@@ -0,0 +1,673 @@
+                    });
+
+                    const logos = parsedContent.logo;
+                    if (logos) {
+                      logos.forEach((logo) => {
+                        const { title, offset } = logo;
+                        const url = getURLForPicto(title);
+                        if (url) {
+                          const { begin, end } = offset;
+                          const overlay = {
+                            start : video.startTime + (begin || 0),
+                            end : video.endTime - (end || 0),
+                            timescale : 1,
+                            version : 1,
+                            elements : [{
+                              url,
+                              format : "png",
+                              xAxis : "0%",
+                              yAxis : "0%",
+                              height : "100%",
+                              width : "100%",
+                            }],
+                          };
+                          const logoStart = video.startTime;
+                          const logoEnd = logoStart + 10;
+                          overlay.start = logoEnd;
+                          const overlayData = JSON.parse(
+                            JSON.stringify(overlay)
+                          );
+                          overlayData.elements.push({
+                            url: channelLogoURL,
+                            format: "png",
+                            xAxis: "85%",
+                            yAxis: "10%",
+                            height: "10%",
+                            width: "12%",
+                          });
+                          overlayData.start = logoStart;
+                          overlayData.end = logoEnd;
+                          logoOverlays.push(overlayData);
+                          overlays.push(overlay);
+                        }
+                      });
+                    } else {
+                      logoOverlays.push({
+                        start: video.startTime,
+                        end: video.startTime + 20,
+                        timescale: 1,
+                        version: 1,
+                        elements: [
+                          {
+                            url: channelLogoURL,
+                            format: "png",
+                            xAxis: "85%",
+                            yAxis: "10%",
+                            height: "10%",
+                            width: "12%",
+                          },
+                        ],
+                      });
+                    }
+                    getKID(manifestURL).then(({ kid }) => {
+                      kidByAffairePgrm[kid] = licenseURL;
+                      resolve();
+                    }).catch((error) => {
+                      reject(error);
+                    });
+                  };
+                  xhr.send();
+
+                }).catch(() => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define this function outside of a loop. ([javascript:FunctionDefinitionInsideLoop](https://sonarcloud.io/organizations/rx-player/rules#rule_key=javascript%3AFunctionDefinitionInsideLoop))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=366&issues=AWbt7Z5cW3Y69bz5aGXT&open=AWbt7Z5cW3Y69bz5aGXT)

[ISSUE_KEY]: AWbt7Z5cW3Y69bz5aGXT